jvb-api/jvb-api-server/src/main/kotlin/org/jitsi/videobridge/api/server/v1/Colibri.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge.api.server.v1
+
+import io.ktor.http.cio.websocket.Frame
+import io.ktor.http.cio.websocket.WebSocketSession
+import io.ktor.http.cio.websocket.readText
+import io.ktor.routing.Route
+import io.ktor.websocket.DefaultWebSocketServerSession
+import io.ktor.websocket.webSocket
+import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.channels.produce
+import org.jitsi.videobridge.api.util.SmackXmlSerDes
+import org.jitsi.xmpp.extensions.colibri.ColibriConferenceIQ
+import org.jitsi.xmpp.extensions.health.HealthCheckIQ
+import org.jivesoftware.smack.packet.IQ
+import org.jivesoftware.smack.packet.Stanza
+import org.jivesoftware.smack.packet.XMPPError
+
+/**
+ * Handles incoming Colibri messages over HTTP websocket
+ */
+fun Route.colibriApi() {
+    webSocket("ws") {
+        for (stanza in incomingStanzas) {
+            if (stanza is IQ) {
+                when (stanza) {
+                    is HealthCheckIQ -> {
+                        val result = call.confManager.handleHealthIq(stanza)
+                        result.stanzaId = stanza.stanzaId
+                        sendStanza(result)
+                    }
+                    is ColibriConferenceIQ -> {
+                        val result = call.confManager.handleColibriConferenceIQ(stanza)
+                        result.stanzaId = stanza.stanzaId
+                        sendStanza(result)
+                    }
+                    else -> {
+                        sendStanza(IQ.createErrorResponse(stanza, XMPPError.Condition.bad_request))
+                    }
+                }
+            }
+            // TODO: other messages we just ignore, can we generate error
+            // responses for Message & Presence?
+        }
+    }
+}
+
+/**
+ * Parse the incoming websocket [Frame]s into [IQ]s.
+ */
+private val WebSocketSession.incomingStanzas: ReceiveChannel<Stanza>
+    get() = produce {
+        for (msg in incoming) {
+            when (msg) {
+                is Frame.Text -> {
+                    val stanza = SmackXmlSerDes.deserialize(msg.readText())
+                    send(stanza)
+                }
+                else -> throw IllegalStateException("Expected only text frames")
```

Comment:
It will...this would basically be a programming bug so the idea was to fail fast so it could be caught and fixed quickly.

Reply:
Well, this could be a programming error but it could also be an attack (someone connects to your ws port and sends a binary data frame).  If it just kills that ws connection that's ideal, but if it causes anything else to go wrong that's a problem.

Reply:
It won't affect the other websocket sessions.  I wrote a test to verify this, but I'm not sure it makes sense to actually commit.

Reply:
I did tweak this line a bit which made the test nicer--I think it won't matter in a real session but pushed it anyway.